cli/src/main.rs

```diff
@@ -192,6 +195,11 @@ async fn run_upload(
 
     let (file_sets, file_counter) =
         build_filesets(&repo, &junit_paths, team.clone(), &codeowners, exec_start)?;
+
+    if (!allow_missing_junit_files && file_counter.get_count() == 0) || file_sets.is_empty() {
```

Comment:
I think this logic needs to be switched around, no?

```suggestion
    if !allow_missing_junit_files && (file_counter.get_count() == 0 || file_sets.is_empty()) {
```

Reply:
Good catch